src/server.ts

```diff
@@ -408,13 +407,19 @@ export class Server {
     transaction: Transaction | FeeBumpTransaction,
     networkPassphrase?: string,
   ): Promise<Transaction | FeeBumpTransaction> {
-    const [{ passphrase }, { footprint }] = await Promise.all([
+    const [{ passphrase }, { error, results }] = await Promise.all([
       networkPassphrase
         ? Promise.resolve({ passphrase: networkPassphrase })
         : this.getNetwork(),
       this.simulateTransaction(transaction),
     ]);
-    return addFootprint(transaction, passphrase, footprint);
+    if (error) {
+      throw error;
+    }
+    if (!results) {
+      throw new Error("transaction simulation failed");
+    }
+    return prepareTransaction(transaction, passphrase, results);
```

Comment:
at first I read this as recursive, just for minor consideration, alias or rename the imported `prepareTransaction` to something like `assembleTransaction`